mmedit/models/backbones/sr_backbones/basicvsr.py

```diff
@@ -0,0 +1,417 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule
+from mmcv.runner import load_checkpoint
+
+from mmedit.models.common import (PixelShufflePack, ResidualBlockNoBN,
+                                  flow_warp, make_layer)
+from mmedit.models.registry import BACKBONES
+from mmedit.utils import get_root_logger
+
+
+@BACKBONES.register_module()
+class BasicVSR(nn.Module):
+    """BasicVSR network structure for video super-resolution.
+
+    Support only x4 upsampling.
+    Paper:
+        BasicVSR: The Search for Essential Components in Video Super-Resolution
+        and Beyond, CVPR, 2021
+
+    Args:
+        mid_channels (int): Channel number of the intermediate features.
+            Default: 64.
+        num_blocks (int): Number of residual blocks in each propagation branch.
+            Default: 30.
+        spynet_pretrained (str): Pre-trained model path of SPyNet.
+            Default: None.
+    """
+
+    def __init__(self, mid_channels=64, num_blocks=30, spynet_pretrained=None):
```

Comment:
Is `spynet_pretrained ` a hard requirement here? 
It seems that `SPyNet` is a sub-model of `BasicVSR` and its parameters cannot be saved separately.

Reply:
When training BasicVSR, we assume the optical flow network is pretrained. Therefore, we need to separately initialize the weights of SPyNet even if BasicVSR is trained from scratch. Pretrained weights will be provided.

Reply:
Fine.